scripts/common.js

```diff
@@ -647,8 +647,185 @@ export function extractTopicsAndProducts() {
   }
 }
 
+/**
+ * Set up click event on Region Picker Feds
+ * Builds Dropdown Based on Region List
+ * Get selected region from Dropdown Selector
+ */
+function handleDropdownRegion() {
+  const regionsNameList = [
+    {
+      lang: 'en_apac',
+      name: 'APAC (English)',
+      home: `/en/apac.html`,
+    },
+    {
+      lang: "ko",
+      name: "Korea (한국어)",
+      home: `/ko/`,
+    },
+    {
+      lang: "en_uk",
+      name: "UK (English)",
+      home: `/en/uk.html`,
+    },
+    {
+      lang: "en",
+      name: "US (English)",
+      home: `/`,
+    }
+  ];
+
+  // Add Region Dropdown Container before Feds Footer
+  const fedsFooter = document.querySelector('#feds-footer');
+  if (fedsFooter) {
+    const regionDropdownContainer = document.createElement('div');
+    regionDropdownContainer.classList.add('region-dropdown');
+    regionDropdownContainer.innerHTML = `<ul class="region-dropdown-list"></ul>`;
+    fedsFooter.parentElement.insertBefore(regionDropdownContainer, fedsFooter);
+  }
+  
+  // Get actual selected Region
+  const getSelectedRegion = () => {
+    const regionPage = regionsNameList.find(r => r.home === location.pathname);
+    // check if Region name matches the blog.language example 'en' in order to show the Region Name in the buttom
+    const langPage = regionsNameList.find(r => r.lang === window.blog.language);
+    let regionLang;
+    let regionName;
+    if (!regionPage) {
+      // region array is empty, we are not on a region page -> check the sessionStorage, if no value, use blog.language
+      regionLang = sessionStorage.getItem('blog-selected-language') || window.blog.language;
+      // in order to show the Region name either from SessionStorage language or blog.language in our region button
+      if (regionLang !== window.blog.language) {
+        // if the selected Region lang do not match the blog.language we get the Region name based on the sessionStorage saved Language Value
+        const storedLanguage = regionsNameList.find(r => r.lang === regionLang);
+        regionName = storedLanguage.name;
```

Comment:
Maybe you can explore storing both language and name in Session Storage instead of just the language. It would enable you to get the region name directly instead of going through `find`